js/src/utils/createErrorMessageForRejectedPromises.test.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Internal dependencies
+ */
+import createErrorMessageForRejectedPromises from './createErrorMessageForRejectedPromises';
+import createMessageForMultipleErrors from '.~/utils/createMessageForMultipleErrors';
+
+jest.mock( '.~/utils/createMessageForMultipleErrors', () => jest.fn() );
+
+describe( 'createErrorMessageForRejectedPromises', () => {
+	const successPromise = () => new Promise( ( resolve ) => resolve( 'OK' ) );
+	const rejectedPromise = () =>
+		new Promise( ( r, reject ) => reject( 'Failed' ) );
+
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'No rejected Promises', async () => {
+		const promises = [
+			successPromise(),
+			successPromise(),
+			successPromise(),
+		];
+		const promisesName = [ 'Promise A', 'Promise B', 'Promise C' ];
+
+		await createErrorMessageForRejectedPromises( promises, promisesName );
+
+		expect( createMessageForMultipleErrors ).toBeCalledTimes( 1 );
+		expect( createMessageForMultipleErrors ).toBeCalledWith( [], true );
+	} );
+
+	it( 'One rejected Promise', async () => {
+		const promises = [
+			successPromise(),
+			rejectedPromise(),
+			successPromise(),
+		];
+		const promisesName = [ 'Promise A', 'Promise B', 'Promise C' ];
+
+		await createErrorMessageForRejectedPromises( promises, promisesName );
+
+		expect( createMessageForMultipleErrors ).toBeCalledTimes( 1 );
+		expect( createMessageForMultipleErrors ).toBeCalledWith(
+			[ 'Promise B' ],
+			true
+		);
+	} );
+
+	it( 'Multiple rejected promises', async () => {
+		const promises = [
+			successPromise(),
+			rejectedPromise(),
+			rejectedPromise(),
+			rejectedPromise(),
+		];
+		const promisesName = [
+			'Promise A',
+			'Promise B',
+			'Promise C',
+			'Promise D',
+		];
+
+		await createErrorMessageForRejectedPromises( promises, promisesName );
+
+		expect( createMessageForMultipleErrors ).toBeCalledTimes( 1 );
+		expect( createMessageForMultipleErrors ).toBeCalledWith(
+			[ 'Promise B', 'Promise C', 'Promise D' ],
+			true
+		);
+	} );
+
+	it( 'All rejected promises', async () => {
+		const promises = [
+			rejectedPromise(),
+			rejectedPromise(),
+			rejectedPromise(),
+		];
+		const promisesName = [ 'Promise A', 'Promise B', 'Promise C' ];
+
+		await createErrorMessageForRejectedPromises( promises, promisesName );
+
+		expect( createMessageForMultipleErrors ).toBeCalledTimes( 1 );
+		expect( createMessageForMultipleErrors ).toBeCalledWith(
+			[ 'Promise A', 'Promise B', 'Promise C' ],
+			false
+		);
+	} );
```

Comment:
:nail_care: :scroll: 
I think it would be nice to test also the return values of `createErrorMessageForRejectedPromises`, as that is what we use in real scenarios, but do not test here.

Reply:
I tried adding locally but it looks redundant since it tests the almost same test cases in `js/src/utils/createMessageForMultipleErrors.test.js`. Although I agree this test is too specific to the implementation details rather than its inputs and outputs, I'm not quite sure about the reason for adding the same test cases.

Reply:
I added assertions for the return values in https://github.com/woocommerce/google-listings-and-ads/pull/1602/commits/66bb465f0cdaf0420dff2d51a95a064063cf7e9b.

Reply:
Thanks. Alternatively, I was thinking to test it like 
```js
it( 'Called with one rejected promise, returns the value of `createMessageForMultipleErrors` called with its name, and `true`' 
...
expect( message ).toEqual( createMessageForMultipleErrors( [ 'Promise B' ],	true ) );
```

It still describes more of an implementation detail, but is less redundant and fragile to `createMessageForMultipleErrors` and message changes.

Reply:
Thanks for the suggestion. This approach looks better. I reverted the https://github.com/woocommerce/google-listings-and-ads/commit/66bb465f0cdaf0420dff2d51a95a064063cf7e9b and adopted this in https://github.com/woocommerce/google-listings-and-ads/pull/1602/commits/2426b2af44f3c1028991d5abd28525103043d5b6.